persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java

```diff
@@ -52,85 +63,106 @@ public class Init implements BLauncherCmd {
     private final PrintStream outStream = System.out;
     private final String configPath = PersistToolsConstants.CONFIG_SCRIPT_FILE;
 
-    private String name = "";
+    private String configName = "";
+    private String projectName = "";
     public String sourcePath = "";
     private static final String COMMAND_IDENTIFIER = "persist-init";
 
+
     Project balProject;
 
     @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
     private boolean helpFlag;
 
     @Override
     public void execute() {
+        Path persistTomlPath = Paths.get(this.sourcePath, SUBMODULE_PERSIST, PERSIST_TOML_FILE);
+        Path databaseConfigPath = Paths.get(this.sourcePath, SUBMODULE_MODULES, KEYWORD_CLIENTS,
+                DATABASE_CONFIGURATION_BAL);
         if (helpFlag) {
             String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
             errStream.println(commandUsageInfo);
             return;
         }
         try  {
             balProject = ProjectLoader.loadProject(Paths.get(sourcePath));
-            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
-                    .descriptor().name().value() + "." + "clients";
+            configName = balProject.currentPackage().descriptor().name().value() + "." + KEYWORD_CLIENTS;
+            projectName = balProject.currentPackage().descriptor().name().value();
         } catch (ProjectException e) {
             errStream.println("Not a Ballerina project (or any parent up to mount point)\n" +
                     "You should run this command inside a Ballerina project. ");
             return;
         }
-        if (!Files.exists(Paths.get(sourcePath, configPath))) {
-            try {
+        File persistToml = new File(persistTomlPath.toString());
+        File databaseConfig = new File(databaseConfigPath.toString());
+        if (persistToml.exists() || databaseConfig.exists()) {
+            errStream.println("`bal persist init` command can only be used once to initialize the project");
+            return;
+        }
+        Generate generateCMD = new Generate();
+        generateCMD.setSourcePath(Paths.get(sourcePath).toAbsolutePath().toString());
+        try {
+            generateConfigurationBalFile();
+            outStream.println("Created database_configuration.bal file with configurations.");
+            createPersistToml(persistTomlPath);
+            outStream.println("Created Persist.toml file with configurations.");
+            generateCMD.execute();
```

Comment:
generate command invocation should be moved after the configuration file is updated. Whether config file is updated or not should not depend on whether generate succeeds

Reply:
updated in [3bfce84](https://github.com/ballerina-platform/persist-tools/pull/52/commits/3bfce8448ba5b238d99c06896cf50acf076134e8)